### ISSUE TITLE
Support error format "detailed" and custom

### ISSUE THREAD

**Issue by andresaraujo** [brion25/jc-react-validate#2]
This will allow to handle custom reporting of errors. An example of this is the password constraints, as they are reported like a check list.

```javascript
const options = {
  inspect: {
    on: 'state'
  },
  format: validatorFormat, // this will only show validator type as an error (custom)
  constraints: {
    email: emailConstraints
  }
}

//so the state after validations will look like:
{
  _validate: {
    email: ['length', 'containsNumber', 'containsUpperLowerLetter', 'containsSymbol']
  }
}

```

See:
https://validatejs.org/#validate-error-formatting

**Comment by brion25** [comment]
I have that on the plate for the release `0.0.2` 😄

**Comment by brion25** [comment]
this is fix on #4
